File: src/main/java/de/moritzf/quota/UsageWindow.java
package de.moritzf.quota;

import java.time.Instant;

/**
 * Represents usage information for one quota window.
 */
public class UsageWindow {
    private double usedPercent;
    private Integer windowMinutes;
    private Instant resetsAt;

    public double getUsedPercent() {
        return usedPercent;
    }

    public void setUsedPercent(double usedPercent) {
        this.usedPercent = usedPercent;
    }

    public Integer getWindowMinutes() {
        return windowMinutes;
    }

    public void setWindowMinutes(Integer windowMinutes) {
        this.windowMinutes = windowMinutes;
    }

    public Instant getResetsAt() {
        return resetsAt;
    }

    public void setResetsAt(Instant resetsAt) {
        this.resetsAt = resetsAt;
    }

    @Override
    public String toString() {
        return "UsageWindow{" +
                "usedPercent=" + usedPercent +
                ", windowMinutes=" + windowMinutes +
                ", resetsAt=" + resetsAt +
                '}';
    }
}

File: src/main/kotlin/de/moritzf/quota/openai/UsageWindow.kt
package de.moritzf.quota.openai

import kotlinx.datetime.Instant
import java.time.Duration

/**
 * Represents usage information for one quota window.
 */
class UsageWindow(
    var usedPercent: Double = 0.0,
    var windowDuration: Duration? = null,
    var resetsAt: Instant? = null,
) {
    override fun toString(): String {
        return "UsageWindow(usedPercent=$usedPercent, windowDuration=$windowDuration, resetsAt=$resetsAt)"
    }
}
